runtime/lua/vim/lsp.lua

```diff
@@ -1180,10 +1192,7 @@ function lsp.start_client(config)
   ---@see vim.lsp.rpc.client_errors for possible errors. Use
   ---`vim.lsp.rpc.client_errors[code]` to get a human-friendly name.
   function dispatch.on_error(code, err)
-    if log.error() then
-      log.error(log_prefix, 'on_error', { code = lsp.client_errors[code], err = err })
-    end
-    err_message(log_prefix, ': Error ', lsp.client_errors[code], ': ', vim.inspect(err))
+    write_error(code, err)
     if config.on_error then
       local status, usererr = pcall(config.on_error, code, err)
```

Comment:
How can we document this behavior? The three (or four) additional codes are custom extensions that is not included in `vim.lsp.rpc`.

FYI, the current documentation about `on_error` `|vim.lsp.start_clinet()|` 

```
                  • on_error: Callback with parameters (code, ...), invoked
                    when the client operation throws an error. `code` is a
                    number describing the error. Other arguments may be passed
                    depending on the error kind. See
                    `vim.lsp.rpc.client_errors` for possible errors. Use
                    `vim.lsp.rpc.client_errors[code]` to get human-friendly
                    name.
```

Reply:
I'm aware of the documentation of `on_error`, and the way I understand it that function is used when there's an issue once the RPC connection has been established. In this PR I'm dealing with errors that happen before even starting the client, so I'm not sure if that same handler should be used here as well. As for the error codes, there is already an extension to the client codes on master today, but I do agree that they deserve further documentation.